SU2_CFD/src/variable_direct_mean_inc.cpp

```diff
@@ -0,0 +1,469 @@
+  
+  bool low_fidelity = config->GetLowFidelitySim();
+  bool dual_time = ((config->GetUnsteady_Simulation() == DT_STEPPING_1ST) ||
+                    (config->GetUnsteady_Simulation() == DT_STEPPING_2ND));
+  bool windgust = config->GetWind_Gust();
+  
+  /*--- Array initialization ---*/
+  
+  TS_Source = NULL;
+  Primitive = NULL;
+  
+  Gradient_Primitive = NULL;
+  
+  Limiter_Primitive = NULL;
+  
+  WindGust    = NULL;
+  WindGustDer = NULL;
+  
+  nPrimVar     = 0;
+  nPrimVarGrad = 0;
+  
+  nSecondaryVar     = 0;
+  nSecondaryVarGrad = 0;
+
+  Undivided_Laplacian = NULL;
+
+  /*--- Allocate and initialize the primitive variables and gradients ---*/
+  
+  nPrimVar = nDim+5; nPrimVarGrad = nDim+3;
+
+	/*--- Allocate residual structures ---*/
+  
+	Res_TruncError = new su2double [nVar];
+  
+	for (iVar = 0; iVar < nVar; iVar++) {
+		Res_TruncError[iVar] = 0.0;
+	}
+  
+	/*--- Only for residual smoothing (multigrid) ---*/
+  
+	for (iMesh = 0; iMesh <= config->GetnMGLevels(); iMesh++)
+		nMGSmooth += config->GetMG_CorrecSmooth(iMesh);
+  
+  if ((nMGSmooth > 0) || low_fidelity) {
+		Residual_Sum = new su2double [nVar];
+		Residual_Old = new su2double [nVar];
+	}
+  
+	/*--- Allocate undivided laplacian (centered) and limiter (upwind)---*/
+  
+	if (config->GetKind_ConvNumScheme_Flow() == SPACE_CENTERED) {
+		Undivided_Laplacian = new su2double [nVar];
+  }
+  
+  /*--- Always allocate the slope limiter,
+   and the auxiliar variables (check the logic - JST with 2nd order Turb model - ) ---*/
+  
+  Limiter_Primitive = new su2double [nPrimVarGrad];
+  for (iVar = 0; iVar < nPrimVarGrad; iVar++)
+    Limiter_Primitive[iVar] = 0.0;
+
+  Limiter = new su2double [nVar];
+  for (iVar = 0; iVar < nVar; iVar++)
+    Limiter[iVar] = 0.0;
+  
+  Solution_Max = new su2double [nPrimVarGrad];
+  Solution_Min = new su2double [nPrimVarGrad];
+  for (iVar = 0; iVar < nPrimVarGrad; iVar++) {
+    Solution_Max[iVar] = 0.0;
+    Solution_Min[iVar] = 0.0;
+  }
+  
+	/*--- Solution and old solution initialization ---*/
+
+  Solution[0] = val_pressure;
+  Solution_Old[0] = val_pressure;
+  for (iDim = 0; iDim < nDim; iDim++) {
+    Solution[iDim+1] = val_velocity[iDim]*config->GetDensity_FreeStreamND();
+    Solution_Old[iDim+1] = val_velocity[iDim]*config->GetDensity_FreeStreamND();
+  }
+
+  
+	/*--- Allocate and initialize solution for dual time strategy ---*/
+  
+	if (dual_time) {
+    Solution_time_n[0]  =  val_pressure;
+    Solution_time_n1[0] =  val_pressure;
+    for (iDim = 0; iDim < nDim; iDim++) {
+      Solution_time_n[iDim+1] = val_velocity[iDim]*config->GetDensity_FreeStreamND();
+      Solution_time_n1[iDim+1] = val_velocity[iDim]*config->GetDensity_FreeStreamND();
+    }
+	}
+  
+	/*--- Allocate space for the time spectral source terms ---*/
+  
+  if (config->GetUnsteady_Simulation() == HARMONIC_BALANCE) {
+		TS_Source = new su2double[nVar];
+		for (iVar = 0; iVar < nVar; iVar++) TS_Source[iVar] = 0.0;
+	}
+    
```

Comment:
Also the comment should be /*--- Allocate space for harmonic balance source terms